### DISCUSSION TITLE
Is there a way to pause a loop()?

### DISCUSSION THREAD

**Discussion by oaklandgit** [replit/kaboom#413] [Q&A] [answered]
Hi,
I coded my game to pause/unpause when the user presses the "p" key:

```js
keyPress("p", () => {
    every((obj)=> {
        obj.paused = !obj.paused
    })
})
```
but any running `loop()` functions aren't affected by that so, of course, some aspects of the game continue to run.

My questions:

1. Is there a way to tell a `loop()` to pause?
2. If not, suggested syntax:

```js
const myLoop = loop(3, () => {
    // do something every 3 seconds
})

myLoop.paused = true //  <-- same format as pausing object updates
```

Short of this, the only way I can think of is to cancel the loop and then re-invoke it.

**Comment by slmjkdbtl** [comment, marked as answer]
There's no pause control at the moment. You can use
```js
let paused = false

loop(3, () => {
    if (paused) return
    // ...
})
```
To achieve pausing, however it's pausing the invocation not the timer.

Currently `loop()` returns a function that cancels the loop like all event handlers. It's a bit weird to have a field under a function, but I like the syntax to treat the returned thing as a general controller instead of just a function. So maybe 
```js
const myLoop = loop()

myLoop.stop()
myLoop.paused = true
```
instead of 
```js
const stopLoop = loop()
stopLoop()
```
in future versions

**Reply by oaklandgit** [reply]
Got it. Thanks @slmjkdbtl !
